pandas/io/excel.py

```diff
@@ -425,6 +426,13 @@ def __init__(self, io, **kwds):
             raise ValueError('Must explicitly set engine if not passing in'
                              ' buffer or path for io.')
 
+        if should_close:
```

Comment:
can you move this to a function in pandas.io.common, call it

``maybe_close_filepath(should_close, io)`` to make this code more conscise